src/test/versionbits_tests.cpp

```diff
@@ -98,60 +107,25 @@ class VersionBitsTester
         return *this;
     }
 
-    VersionBitsTester& TestDefined() {
+    VersionBitsTester& TestState(ThresholdState exp) {
         for (int i = 0; i < CHECKERS; i++) {
             if (InsecureRandBits(i) == 0) {
-                BOOST_CHECK_MESSAGE(checker[i].GetStateFor(vpblock.empty() ? nullptr : vpblock.back()) == ThresholdState::DEFINED, strprintf("Test %i for DEFINED", num));
-                BOOST_CHECK_MESSAGE(checker_always[i].GetStateFor(vpblock.empty() ? nullptr : vpblock.back()) == ThresholdState::ACTIVE, strprintf("Test %i for ACTIVE (always active)", num));
+                ThresholdState got = checker[i].GetStateFor(vpblock.empty() ? nullptr : vpblock.back());
+                ThresholdState got_always = checker_always[i].GetStateFor(vpblock.empty() ? nullptr : vpblock.back());
+                int height = vpblock.empty() ? 0 : (1 + vpblock.back()->nHeight);
```

Comment:
bae9a452191a7a83478f7d508a54f4a04d385505  and bd715ff894344408df2345d9bd75a08bd7c9c6f6 : why the `+ 1`? The log will show `0` for the genesis block, but 2 the block at height 1.

Reply:
`height` here is the height of the next block -- if `vpblock` is empty, the next (ie first) block should be the genesis block with `nHeight == 0`. Note that when populating `vpblock` in `VersionBitsTester::Mine()` we set `nHeight=vblock.size()` so the first entry will have `nHeight == 0` and act as the genesis block; at which point `1+vpblock.back().nHeight` will evaluate to `1+0` giving the height of the next block to be added. When `vpblock` is empty, it'll evaluate to `0`, still giving the height of the next block to be added.

Reply:
Note that `GetStateFor(block)` is telling you what the state should be for the *next* block to be mined, not the last block that was mined.

Reply:
I see, I'll add a comment to that effect in my PR.